### PR TITLE
Implement window move

### PR DESCRIPTION
**Pull request by happenslol**
#### Describe your PR, what does it fix/add?

As discussed in #1999, this addresses the second issue.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

I've refactored the whole monitor focusing thing a bit, so remembering the last workspace is consistent across monitor switches. In the long run I think `changeworkspace("[internal]")` could also use a refactor, pulling that functionality out could remove a lot of duplication here.

I also added `std::ignore`s to a few unused results to suppress the warnings, I can remove those if you want to.

### CODE REVIEW

File: src/helpers/MiscFunctions.cpp [hunk 1]
```diff
@@ -161,7 +161,7 @@ std::string getFormat(const char* fmt, ...) {
 
     va_list args;
     va_start(args, fmt);
-    vasprintf(&outputStr, fmt, args);
+    std::ignore = vasprintf(&outputStr, fmt, args);
```
Comment (vaxerski) [inline src/helpers/MiscFunctions.cpp:164]: I don't believe all of these belong in this PR.

File: src/managers/KeybindManager.cpp [hunk 2]
```diff
@@ -63,7 +64,6 @@ CKeybindManager::CKeybindManager() {
     m_mDispatchers["lockgroups"]                    = lockGroups;
     m_mDispatchers["moveintogroup"]                 = moveIntoGroup;
     m_mDispatchers["moveoutofgroup"]                = moveOutOfGroup;
-    m_mDispatchers["global"]                        = global;
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:66]: ummm... unintentional?

File: src/managers/KeybindManager.cpp [hunk 3]
```diff
@@ -2182,16 +2198,3 @@ void CKeybindManager::moveOutOfGroup(std::string args) {
 
     g_pKeybindManager->m_bGroupsLocked = GROUPSLOCKEDPREV;
 }
-
-void CKeybindManager::global(std::string args) {
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:2186]: again...?

File: src/managers/KeybindManager.cpp [hunk 4]
```diff
@@ -1312,14 +1334,23 @@ void CKeybindManager::moveActiveTo(std::string args) {
...
+    }
 
-    if (!PWINDOWTOCHANGETO)
+    // Otherwise, we always want to move to the next monitor in that direction
+    const auto PMONITORTOCHANGETO = g_pCompositor->getMonitorInDirection(arg);
+    if (!PMONITORTOCHANGETO)
         return;
 
-    g_pLayoutManager->getCurrentLayout()->switchWindows(PLASTWINDOW, PWINDOWTOCHANGETO);
+    if (!trySetActiveMonitor(PMONITORTOCHANGETO, false))
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:1350]: why? moveActiveToWorkspace will focus the workspace. It's not silent.

File: src/managers/KeybindManager.cpp [hunk 5]
```diff
@@ -614,14 +628,14 @@ uint64_t CKeybindManager::spawnRaw(std::string args) {
             _exit(0);
         }
         close(socket[0]);
-        write(socket[1], &grandchild, sizeof(grandchild));
+        std::ignore = write(socket[1], &grandchild, sizeof(grandchild));
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:631]: ditto. Doesn't belong 'ere

**Review by vaxerski — CHANGES_REQUESTED** [review]
amogus

File: src/managers/KeybindManager.cpp (hunk 2 above)
Comment (happenslol) [reply src/managers/KeybindManager.cpp:66]: epic rebase prank gone wrong

File: src/helpers/MiscFunctions.cpp (hunk 1 above)
Comment (happenslol) [reply src/helpers/MiscFunctions.cpp]: > I also added std::ignores to a few unused results to suppress the warnings, I can remove those if you want to.

Alright, reverted. Warnings were annoying me during dev tbh

File: src/managers/KeybindManager.cpp (hunk 4 above)
Comment (happenslol) [reply src/managers/KeybindManager.cpp:1350]: You're right actually, didn't think about that. Makes other things a bit easier too

**Comment by happenslol** [comment]
Updated and tested

ඞ

File: src/managers/KeybindManager.cpp [hunk 6]
```diff
@@ -2166,6 +2172,19 @@ void CKeybindManager::moveIntoGroup(std::string args) {
     PWINDOW->m_dWindowDecorations.emplace_back(std::make_unique<CHyprGroupBarDecoration>(PWINDOW));
 }
 
+void CKeybindManager::global(std::string args) {
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:2175]: why is this moved

**Review by vaxerski — CHANGES_REQUESTED** [review]
nit

File: src/managers/KeybindManager.cpp (hunk 6 above)
Comment (happenslol) [reply src/managers/KeybindManager.cpp]: It's been a long day

Comment (vaxerski) [reply src/managers/KeybindManager.cpp]: long like my dicc hehehehe (just kidding I have a very small penis)

**Review by vaxerski — APPROVED** [review]
lgtm

**Comment by vaxerski** [comment]
if you could make a wiki mr for the new swapactive dispatcher it'd be great

**Comment by happenslol** [comment]
> if you could make a wiki mr for the new swapactive dispatcher it'd be great

Yeah sure I'll have time for that tomorrow, I'll also add the new `no_window_fallback` option to the docs then
